src/iosrv/vos.c

```diff
@@ -785,16 +786,20 @@ enum_unpack_recxs(daos_key_desc_t *kds, void *data,
 		return -DER_INVAL;
 
 	iod = &io->ui_iods[io->ui_iods_size - 1];
-	iod_akey = &iod->iod_name;
+	rc = daos_iov_copy(&iod_akey, &iod->iod_name);
```

Comment:
What is the purpose of copy the iod_name into another independent buffer? any benefit?

Reply:
Because in the following loop, the iod_akey might be enqueued, which might be freed, but we need akey to pack the following record in the loop.